tests/unit-tests/blocks/test-class-sensei-global-blocks.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * Tests for Sensei_Course_Progress_Block class.
+ */
+class Sensei_Global_Blocks_Test extends WP_UnitTestCase {
+
+	/**
+	 * Instance of Sensei Global Blocks
+	 *
+	 * @var Sensei_Global_Blocks
+	 */
+	private $global_blocks;
+
+	/**
+	 * List of blocks
+	 *
+	 * @var Array
+	 */
+	private $blocks;
+
+	/**
+	 * Set up the test.
+	 */
+	public function setUp() {
+		$this->global_blocks = new Sensei_Global_Blocks();
+		$this->blocks        = [
+			'sensei-lms/button-take-course',
+			'sensei-lms/button-view-results',
+			'sensei-lms/course-progress',
+			'sensei-lms/button-continue-course',
+			'sensei-lms/course-categories',
+			'sensei-lms/course-list-filter',
+		];
+
+		parent::setUp();
+	}
+
+	/**
+	 * Register all global blocks
+	 *
+	 * @covers Sensei_Global_Blocks->initialize_blocks
+	 */
+	public function testInitializeBlocks_WhenCalled_ShowsRegisterAllBlocks() {
+		$is_registered = function ( string $block ): void {
+			$this->assertTrue( WP_Block_Type_Registry::get_instance()->is_registered( $block ) );
```

Comment:
When there are multiple assertions in the same test case, I like to add an optional message as the last parameter so that it's easier to identify which of the assertions failed, in case not all of them did.